### PR TITLE
refactor(python): Rename pearson_corr & spearman_rank_corr

### PR DESCRIPTION
**Pull request by josemasar**
-  Renaming the .pearson_corr and .spearman_rank_corr methods to a much shorted .corr, with a method='pearson' / 'spearman'.

Issue:
#6802

### CODE REVIEW

File: py-polars/polars/internals/lazy_functions.py [hunk 1]
```diff
@@ -1375,12 +1346,30 @@ def pearson_corr(a: str | pli.Expr, b: str | pli.Expr, ddof: int = 1) -> pli.Exp
...
+    └─────┘
+
     """
     if isinstance(a, str):
         a = col(a)
     if isinstance(b, str):
         b = col(b)
-    return pli.wrap_expr(pypearson_corr(a._pyexpr, b._pyexpr, ddof))
+    if method == "pearson":
+        return pli.wrap_expr(pypearson_corr(a._pyexpr, b._pyexpr, ddof))
```
Comment (zundertj) [inline py-polars/polars/internals/lazy_functions.py:1368]: Does this method not support `propagate_nans`? That is not documented in the docstring currently, although I guess the best course of action is to support that actually.

File: py-polars/polars/internals/lazy_functions.py [hunk 2]
```diff
@@ -1305,13 +1306,15 @@ def cumsum(
     return cumfold(lit(0).cast(UInt32), lambda a, b: a + b, column).alias("cumsum")
 
 
-def spearman_rank_corr(
-    a: str | pli.Expr, b: str | pli.Expr, ddof: int = 1, propagate_nans: bool = False
+def corr(
```
Comment (zundertj) [inline py-polars/polars/internals/lazy_functions.py:1309]: We need to keep the `spearman_rank_corr` and `pearson_corr` methods, but raise a deprecation warning. Right now, users code will erro.

File: py-polars/polars/internals/dataframe/frame.py [hunk 3]
```diff
@@ -7407,7 +7407,7 @@ def unnest(self, names: str | Sequence[str]) -> Self:
         return self._from_pydf(self._df.unnest(names))
 
     @typing.no_type_check
-    def pearson_corr(self, **kwargs: Any) -> Self:
```
Comment (zundertj) [inline py-polars/polars/internals/dataframe/frame.py:7410]: As on expressions, we need to keep this name and raise a deprecation warning.

File: py-polars/polars/internals/lazy_functions.py [hunk 4]
```diff
@@ -1305,13 +1306,15 @@ def cumsum(
...
+    a: str | pli.Expr,
+    b: str | pli.Expr,
+    method: CorrelationMethod = "pearson",
+    ddof: int = 1,
+    propagate_nans: bool = False,
```
Comment (zundertj) [inline py-polars/polars/internals/lazy_functions.py:1314]: `method`, `ddof` and `propogate_nans` should be keyword only imo, but I guess that is what @stinodego  is working on in another PR?

File: py-polars/tests/unit/test_functions.py [hunk 5]
```diff
@@ -130,8 +130,8 @@ def test_null_handling_correlation() -> None:
 
     out = df.select(
         [
-            pl.pearson_corr("a", "b").alias("pearson"),
-            pl.spearman_rank_corr("a", "b").alias("spearman"),
+            pl.corr("a", "b").alias("pearson"),
+            pl.corr("a", "b", "spearman").alias("spearman"),
```
Comment (zundertj) [inline py-polars/tests/unit/test_functions.py:134]: We should call method as a keyword argument, i.e. `corr("a", "b", method="spearman")`

File: py-polars/tests/unit/test_functions.py [hunk 6]
```diff
@@ -141,9 +141,9 @@ def test_null_handling_correlation() -> None:
     df1 = pl.DataFrame({"a": [None, 1, 2], "b": [None, 2, 1]})
     df2 = pl.DataFrame({"a": [np.nan, 1, 2], "b": [np.nan, 2, 1]})
 
-    assert np.isclose(df1.select(pl.spearman_rank_corr("a", "b")).item(), -1.0)
+    assert np.isclose(df1.select(pl.corr("a", "b", "spearman")).item(), -1.0)
```
Comment (zundertj) [inline py-polars/tests/unit/test_functions.py:144]: idem

**Comment by zundertj** [comment]
Im in favor of this change, as the Spearman rank correlation, although useful, is much less used vs the Pearson one, so it makes sense to hide that behind a kwarg.

File: py-polars/polars/internals/lazy_functions.py (hunk 1 above)
Comment (josemasar) [reply py-polars/polars/internals/lazy_functions.py:1455]: Added clarification in the docstrings.

File: py-polars/polars/internals/lazy_functions.py (hunk 2 above)
Comment (josemasar) [reply py-polars/polars/internals/lazy_functions.py:1395]: The original methods are back with deprecation warnings.

File: py-polars/tests/unit/test_functions.py (hunk 5 above)
Comment (josemasar) [reply py-polars/tests/unit/test_functions.py]: Now it´s a keyword argument.

File: py-polars/polars/internals/lazy_functions.py (hunk 4 above)
Comment (stinodego) [reply py-polars/polars/internals/lazy_functions.py:1400]: Well, this is a new function, so let's get it right straight away. Indeed those should be keyword only.

File: py-polars/polars/internals/lazy_functions.py [hunk 7]
```diff
@@ -1373,16 +1377,88 @@ def pearson_corr(a: str | pli.Expr, b: str | pli.Expr, ddof: int = 1) -> pli.Exp
...
     return pli.wrap_expr(pypearson_corr(a._pyexpr, b._pyexpr, ddof))
 
 
+def corr(
+    a: str | pli.Expr,
+    b: str | pli.Expr,
+    method: CorrelationMethod = "pearson",
+    ddof: int = 1,
+    propagate_nans: bool = False,
+) -> pli.Expr:
```
Comment (stinodego) [inline py-polars/polars/internals/lazy_functions.py:1401]:
```suggestion
def corr(
    a: str | pli.Expr,
    b: str | pli.Expr,
    *,
    method: CorrelationMethod = "pearson",
    ddof: int = 1,
    propagate_nans: bool = False,
) -> pli.Expr:
```

Comment (josemasar) [reply py-polars/polars/internals/lazy_functions.py]: I included these suggestions. Thanks!

**Comment by josemasar** [comment]
@stinodego @zundertj I´ve got some trouble resolving the merge conflicts. Can you please help me on this?

**Comment by stinodego** [comment]
I don't think I can solve it for you, since you're on a fork. I'd recommend updating your fork and then do an interactive rebase on master, only selecting the commits you need.

**Comment by josemasar** [comment]
@stinodego Thanks for the tip! The PR is now ready for the final review.

File: py-polars/polars/internals/lazy_functions.py [hunk 8]
```diff
@@ -1376,15 +1380,85 @@ def pearson_corr(a: str | pli.Expr, b: str | pli.Expr, ddof: int = 1) -> pli.Exp
...
+        a = col(a)
+    if isinstance(b, str):
+        b = col(b)
+    if method == "pearson":
+        return pli.wrap_expr(pypearson_corr(a._pyexpr, b._pyexpr, ddof))
+    elif method == "spearman":
+        return pli.wrap_expr(
+            pyspearman_rank_corr(a._pyexpr, b._pyexpr, ddof, propagate_nans)
+        )
+
```
Comment (stinodego) [inline py-polars/polars/internals/lazy_functions.py:1460]: This needs an `else` case raising a ValueError with a standardized message. See `is_between` for what the msg should look like.

File: py-polars/polars/internals/type_aliases.py [hunk 9]
```diff
@@ -71,6 +71,7 @@
 RollingInterpolationMethod: TypeAlias = Literal[
     "nearest", "higher", "lower", "midpoint", "linear"
 ]  # QuantileInterpolOptions
+CorrelationMethod: TypeAlias = Literal["pearson", "spearman"]  # CorrelationOptions
```
Comment (stinodego) [inline py-polars/polars/internals/type_aliases.py:74]: This should be in the block below, under things that do not have an equivalent on the Rust side.

Comment (josemasar) [reply py-polars/polars/internals/type_aliases.py]: @stinodego , I moved `CorrelationMethod` to the code block below.

File: py-polars/polars/internals/lazy_functions.py (hunk 8 above)
Comment (josemasar) [reply py-polars/polars/internals/lazy_functions.py:1465]: @stinodego , I added the `else` clause.

**Comment by ritchie46** [comment]
Thanks for the PR and the reviews. :+1:

**Comment by stinodego** [comment]
@ritchie46 I don't think this was ready to be merged, the old methods weren't deprecated correctly. I can make a follow-up PR to fix it.

**Pull request merged**